lsp-mode.el

```diff
@@ -232,6 +232,21 @@ It contains all of the clients that are currently regitered.")
   :type 'boolean
   :group 'lsp-mode)
 
+(defcustom lsp-eldoc-enable-hover t
+  "If non-nil, eldoc will display hover info when it is present."
+  :type 'boolean
+  :group 'lsp-mode)
+
+(defcustom lsp-eldoc-enable-signature-help nil
```

Comment:
IMO it would be better to have that t and have let server which has expensive signature help should implement a setting for enabling signature capability. This is how it works in VSCode - by default you do not get signature help for JDT server while you get signature help for JS and Python. Disabling the signature globally will lead to poor feature discoverability.

Reply:
I disagree. In vscode signatureHelp is used in combination with completion. It is not triggered so frequently as lsp-mode currently does.

Reply:
We can add evil integration in the future